src/scripts/main.js

```diff
@@ -1,3 +1,280 @@
 'use strict';
 
-// write code here
+// #region sorting Table
+const tableNames = document.querySelectorAll('thead th');
+let lastSortedIndex = null;
+const sortDirections = Array.from({ length: tableNames.length }, () => true);
+
+tableNames.forEach((element, index) => {
+  element.addEventListener('click', () => sortTable(index));
+});
+
+function sortTable(index) {
+  const tableBody = document.querySelector('tbody');
+  const headRows = Array.from(tableBody.querySelectorAll('tr'));
+
+  if (lastSortedIndex === index) {
+    sortDirections[index] = !sortDirections[index];
+  } else {
+    sortDirections[index] = true;
+  }
+
+  headRows.sort((rowA, rowB) => {
+    const cellA = rowA.cells[index].textContent.trim();
+    const cellB = rowB.cells[index].textContent.trim();
+
+    const direction = sortDirections[index] ? 1 : -1;
+
+    return compareValues(cellA, cellB, index) * direction;
+  });
+
+  tableBody.append(...headRows);
+  lastSortedIndex = index;
+}
+
+function compareValues(cellA, cellB, index) {
+  if (index <= 2) {
+    return cellA.localeCompare(cellB);
+  }
+
+  if (index === 4) {
+    const valueA = parseFloat(cellA.replace(/[$,]/g, '')) || 0;
+    const valueB = parseFloat(cellB.replace(/[$,]/g, '')) || 0;
+
+    return valueA - valueB;
+  }
```

Comment:
The condition `index === 4` assumes that the fifth column contains currency values. Ensure this assumption aligns with your table structure. If the table structure changes, this condition might need updating.